samples/subsys/nvs/src/main.c

```diff
@@ -216,6 +216,13 @@ void main(void)
 				}
 				sys_reboot(0);
 			}
+		} else {
+			printk("Reboot counter reached max value.\n");
+			printk("Reset to 0 and exit test.\n");
+			reboot_counter = 0;
+			nvs_write(&fs, RBT_CNT_ID, &reboot_counter,
+			  sizeof(reboot_counter));
+			break;
 		}
 	}
 }
```

Comment:
Not a bug fix, but a nice addition. The disadvantage is that it resets the counter and if you reload it will restart the incrementation, before it would just boot and show the max value. I do agree with the change.